keps/sig-auth/2579-pod-isolation-policy/README.md

```diff
@@ -0,0 +1,951 @@
+2. Policy level must be one of: `privileged`, `baseline`, `restricted`
+3. Version values must be match `(latest|v[0-9]+\.[0-9]+`. That is, one of:
+    1. `latest`
+    2. `vMAJOR.MINOR` (e.g. `v1.21`)
+
+### Versioning
+
+A specific version can be supplied for each enforcement mode. The version pins the policy to the
+version that was defined at that kubernetes version. The default version is `latest`, which can be
+provided to explicitly use the latest definition of the policy. There is some nuance to how versions
+other than latest are applied:
+
+- If the constrained pod field has not changed since the pinned version, the policy is applied as
+  originally specified.
+- The privileged profile always means fully unconstrained and is effectively unversioned (specifying
+  a version is allowed but ignored).
+- Specifying a version more recent than the current Kubernetes version is allowed (for rollback &
+  version skew reasons), but is treated as `latest`.
+
+<<[UNRESOLVED]>>
+
+Under the webhook implementation, if the webhook version is newer than the cluster version, how
+should policy versions past the cluster version be handled? Should `latest` mean the latest version
+supported by the webhook, or the policy version matching the cluster version? _Leaning towards
+supporting all policy versions supported by the webhook, even if they are newer than the cluster
+version._
+
+Note that policies are not guaranteed to be backwards compatible, and a newer restricted policy
+could require setting a field that doesn't exist in the current API version.
+
+<<[/UNRESOLVED]>>
+
+<<[UNRESOLVED]>>
+
+- Under an older version of a policy, a new field may only be set to its default value (or left unset).
+
+**Open Question:** Should this apply to all new fields, or only new fields that are constrained
+under a more recent version of the policy? If we only constrain fields that are in-scope for policy
+enforcement, then that becomes a problem if we later decide that a field should be in scope. The
+versioning machanism gives us a way to safely make "breaking" changes like adding enforcement to a
+previously unenforced field, but it is strictly less safe. Either way, there is some maintenance
+burden to declare which fields are in-scope at each version. _Leaning towards only restricting
+relevant fields._
+
+**Open Question:** If the control (or options) are strictly less privileged than the default, should
+it be allowed? For example, the `allowPrivilegeEscalation` field was added in Kubernetes v1.8, which
+sets the `no_new_privs` flag on the container processes when set to false. Prior to that version,
+the `no_new_privs` flag was never set, which is also the default in v1.8+. If older policies require
+the default value, then only `allowPrivilegeEscalation: true` would be allowed, even though
+`allowPrivilegeEscalation: false` is strictly less privileged. _Leaning towards allowing strictly
+more-restricted versions._
```

Comment:
consensus during breakout meeting: yes, less privileged values should be allowed; e.g. `restricted@v1.7` applied by newer API server should permit the `allowPrivilegeEscalation` field to be:
* null (allowed during v1.7 release)
* true (equal in privilege to a v1.7 pod that didn't set the field)
* false (strictly less privileged than other allowed values)